### ISSUE TITLE
DELETE/UPDATE with limit

### ISSUE THREAD

**Issue by ehmo** [citusdata/citus#1952]
I tried to look this up in the issues, but wasn't successful so hopefully I'm not duplicating anything.

We have multiple uses cases where we need to be able to use `DELETE/UPDATE` with limit. This is traditionally achieved in postgres by running a subquery e.g.
```sql
DELETE FROM table WHERE id in (SELECT id FROM TABLE limit 100) RETURNING id;
```
which can't be currently achieved.

I understand that it's (probably) impossible to distribute the query properly to the shards, but maybe some would be ok with a compromise, where each limit would be multiplied by the number of shards. At least for us this would work well.

Our current workaround is to run query on each shard separately and gather the results on the master and then push them to the pipeline.

**Comment by byucesoy** [comment]
With 7.2, we are introducing some pull-push logic but it focuses on SELECT queries. There is no decided timeline to implement similar logic for modifying queries, but seeing such issues help us during our release planning, so thanks for bringing this to our attention.

Also I'm marking as related with #1947

**Comment by marcocitus** [comment]
Fixed by #2137
